src/nvim/api/buffer.c

```diff
@@ -392,7 +395,9 @@ void nvim_buf_set_lines(uint64_t channel_id,
   // Only adjust marks if we managed to switch to a window that holds
   // the buffer, otherwise line numbers will be invalid.
   if (save_curbuf.br_buf == NULL) {
-    mark_adjust((linenr_T)start, (linenr_T)(end - 1), MAXLNUM, extra);
+    mark_adjust((linenr_T)start, (linenr_T)(end - 1), MAXLNUM, extra, false);
```

Comment:
Maybe, but "neither doing nor undoing" doesn't seem too far fetched for an undo mode enum.